### ISSUE TITLE
Updating `activation_policy` with mysql breaks a number of dependant resources

### ISSUE THREAD

**Issue by alexberry** [terraform-google-modules/terraform-google-sql-db#370]
### TL;DR

When setting `activation_policy` to `NEVER`, can apply successfully, however when setting back to `ALWAYS` we get failures for both resources local to the module and dependant resources outside the module.

### Expected behavior

1. Would expect database created while database was online to remain in state and not require recreation after startup
2. Would expect the google sql user API to gracefully handle a shutdown database

### Observed behavior

# 1. Module-local breakages

Additional databases attempt to get recreated, even if they already exist and were not deleted from state (targeting owing to breakages outside of the module):
```
$ terraform apply -target module.mpbx_db
... Refreshing state... 

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create
  ~ update in-place

Terraform will perform the following actions:

  # module.mpbx_db.google_sql_database.additional_databases["mpbx_locks"] will be created
  + resource "google_sql_database" "additional_databases" {
      + charset   = "utf8mb4"
      + collation = "utf8mb4_general_ci"
      + id        = (known after apply)
      + instance  = "mpbx-db-a04c"
      + name      = "mpbx_locks"
      + project   = "mpb-platform-sre-917-5372"
      + self_link = (known after apply)
    }

  # module.mpbx_db.google_sql_database.additional_databases["mpbx_spooler"] will be created
  + resource "google_sql_database" "additional_databases" {
      + charset   = "utf8mb4"
      + collation = "utf8mb4_general_ci"
      + id        = (known after apply)
      + instance  = "mpbx-db-a04c"
      + name      = "mpbx_spooler"
      + project   = "mpb-platform-sre-917-5372"
      + self_link = (known after apply)
    }
```
These resources require importing after updating it's activation policy

# 2. External Breakages

Once the database has shutdown when it's activation policy is set to `NEVER`,  standard google-provider resource `google_sql_user` returns a 400 error when refreshing state, because the instance is offline:

```
$ terraform plan
...
google_sql_user.mpbx_db_user: Refreshing state... [id=mpbx/%/mpbx-db-a04c]
module.txns_db.random_id.user-password: Refreshing state... [id=tbdCu1zD6Y8]
module.txns_db.google_sql_database.default[0]: Refreshing state... [id=projects/mpb-platform-sre-917-5372/instances/txns-db-836e/databases/transactions]
null_resource.bootstrap_txns_db[0]: Refreshing state... [id=131845498594269417]
null_resource.database_connection_config[0]: Refreshing state... [id=2027946939532478515]
google_sql_user.txns_db_user: Refreshing state... [id=transaction_service//txns-db-836e]
google_sql_user.data_engineering_txns_db_user: Refreshing state... [id=data_engineering//txns-db-836e]
╷
│ Error: Error when reading or editing SQL User "mpbx" in instance "mpbx-db-a04c": googleapi: Error 400: Invalid request: Invalid request since instance is not running., invalid
│
│   with google_sql_user.mpbx_db_user,
│   on mpbx_db.tf line 153, in resource "google_sql_user" "mpbx_db_user":
│  153: resource "google_sql_user" "mpbx_db_user" {
│
╵
```

### Terraform Configuration

```hcl
locals {
  mpbx_db_username = "mpbx"
}

resource "random_id" "mpbx_db_suffix" {
  byte_length = 2
}

module "mpbx_db" {
  activation_policy = var.default_sql_activation_policy
  source  = "GoogleCloudPlatform/sql-db/google//modules/mysql"
  version = "10.0.0"

  name       = "mpbx-db-${random_id.mpbx_db_suffix.hex}"
  project_id = data.terraform_remote_state.project.outputs.project_id

  region = var.region
  zone   = "${var.region}-b"

  database_version    = "MYSQL_5_6"
  deletion_protection = var.database_deletion_protection

  db_name      = "mpbx"
  db_charset   = "utf8mb4"
  db_collation = "utf8mb4_general_ci"
  database_flags = [
    {
      name  = "sql_mode"
      value = "NO_ENGINE_SUBSTITUTION,STRICT_TRANS_TABLES"
    },
    {
      name  = "log_output"
      value = "FILE"
    },
    {
      name  = "slow_query_log"
      value = "on"
    },
    {
      name  = "long_query_time"
      value = 5
    },
  ]

  additional_databases = [
    {
      name      = "mpbx_spooler"
      charset   = "utf8mb4"
      collation = "utf8mb4_general_ci"
    },
    {
      name      = "mpbx_locks"
      charset   = "utf8mb4"
      collation = "utf8mb4_general_ci"
    },
  ]

  enable_default_user = false

  availability_type = var.cloudsql_availability_type

  maintenance_window_day          = var.cloudsql_maintenance_window_day
  maintenance_window_hour         = var.cloudsql_maintenance_window_hour
  maintenance_window_update_track = var.cloudsql_maintenance_window_update_track

  disk_autoresize = var.cloudsql_disk_autoresize
  tier            = var.mpbx_db_tier
  disk_size       = "750"

  backup_configuration = {
    enabled            = var.cloudsql_backups_enabled
    binary_log_enabled = var.cloudsql_backups_enabled

    start_time                     = var.cloudsql_backup_start_time
    transaction_log_retention_days = var.cloudsql_transaction_log_retention_days
    retained_backups               = var.cloudsql_backup_retention_days
    location = var.cloudsql_backups_enabled ? var.region : null

    retention_unit = "COUNT"
  }

  read_replicas = data.terraform_remote_state.project.outputs.project_stage != "prod" ? [] : [
    {
      name                = "-reporting"
      zone                = "europe-west2-a"
      database_flags      = []
      user_labels         = {}
      disk_autoresize     = true
      encryption_key_name = null
      tier                = var.mpbx_db_tier
      disk_size           = "900"
      disk_type           = "PD_SSD"
      ip_configuration = {
        private_network = data.terraform_remote_state.infrastructure.outputs.network_self_link
        ipv4_enabled        = true
        require_ssl         = false
        allocated_ip_range  = null
        authorized_networks = []
      }
    }
  ]

  ip_configuration = {
    private_network = data.terraform_remote_state.infrastructure.outputs.network_self_link
    ipv4_enabled        = data.terraform_remote_state.project.outputs.project_stage != "prod"
    authorized_networks = []
    require_ssl         = false
    allocated_ip_range  = null
  }
  update_timeout = "30m"
}

resource "random_password" "mpbx_db_password" {
  length  = 16
  special = false
}

resource "google_sql_user" "mpbx_db_user" {
  depends_on = [resource.null_resource.bootstrap_mpbx_db]

  instance = module.mpbx_db.instance_name
  project  = data.terraform_remote_state.project.outputs.project_id

  name     = local.mpbx_db_username
  host     = "%"
  password = random_password.mpbx_db_password.result
}

resource "random_password" "data_engineering_mpbx_db_password" {
  length  = 16
  special = false
}

resource "google_sql_user" "data_engineering_mpbx_db_user" {
  depends_on = [resource.null_resource.bootstrap_mpbx_db]

  instance = module.mpbx_db.instance_name
  project  = data.terraform_remote_state.project.outputs.project_id

  name     = "data_engineering"
  host     = "%"
  password = random_password.data_engineering_mpbx_db_password.result
}

resource "google_secret_manager_secret" "mpbx_app_db_password" {
  project   = data.terraform_remote_state.project.outputs.project_id
  secret_id = "mpbx-app-db-password"

  replication {
    automatic = true
  }
}

resource "google_secret_manager_secret_version" "mpbx_app_db_password" {
  secret      = google_secret_manager_secret.mpbx_app_db_password.id
  secret_data = random_password.mpbx_db_password.result
}

resource "google_secret_manager_secret" "mpbx_data_engineering_db_password" {
  project   = data.terraform_remote_state.project.outputs.project_id
  secret_id = "mpbx-data-engineering-db-password"

  replication {
    automatic = true
  }
}

resource "google_secret_manager_secret_version" "mpbx_data_engineering_db_password" {
  secret      = google_secret_manager_secret.mpbx_data_engineering_db_password.id
  secret_data = random_password.data_engineering_mpbx_db_password.result
}
```

### Terraform Version

```sh
Terraform v1.3.3
on darwin_amd64
+ provider registry.terraform.io/hashicorp/google v4.41.0
+ provider registry.terraform.io/hashicorp/google-beta v4.41.0
+ provider registry.terraform.io/hashicorp/local v2.2.3
+ provider registry.terraform.io/hashicorp/null v3.1.1
+ provider registry.terraform.io/hashicorp/random v3.1.3

Your version of Terraform is out of date! The latest version
is 1.3.4. You can update by downloading from https://www.terraform.io/downloads.html
```
